src/EnergyPlus/UnitarySystem.cc

```diff
@@ -14013,91 +14013,99 @@ namespace UnitarySystems {
                 }
             }
         }
-        if (SolFla == -1) {
-            if (!state.dataGlobal->WarmupFlag) {
-                if (this->warnIndex.m_SensPLRIter < 1) {
-                    ++this->warnIndex.m_SensPLRIter;
-                    ShowWarningError(state,
-                                     format("{} - Iteration limit exceeded calculating part-load ratio for unit = {}", this->UnitType, this->Name));
-                    ShowContinueError(state, format("Estimated part-load ratio  = {:.3R}", (ReqOutput / FullOutput)));
-                    ShowContinueError(state, format("Calculated part-load ratio = {:.3R}", PartLoadFrac));
-                    ShowContinueErrorTimeStamp(state, "The calculated part-load ratio will be used and the simulation continues. Occurrence info:");
-                } else {
+        if (FullOutput > 0.0) {
```

Comment:
Why did we get this far if FullOutput = 0? That's why this is a draft, isn't it.

Reply:
Good question. This is a very large function with lots of if blocks for various coil types with blocks for sensible and latent loads, and these call the residual functions. This was the easiest place to put a check. There's probably some better place(s) to do this check.

Reply:
This is in controlCoolingSystemToSP so I am wondering if FullOutput is not getting set correctly, and why a load variable was even used in this function (because old parents used load control and that code carried over). I have long wanted to change this to pure T and w controls.

Reply:
I just saw this on a defect file. SensibleLoad = false, LatentLoad = true and dehum = CoolReheat.  FullOutput is not set in the CoolReheat block. I did this as a temporary solution until I can look at this more.

    PartLoadFrac = (FullOutput > 0.0) ? ReqOutput / FullOutput : 1.0; // because I assume the coil needs to run, I just don't know how much